packages/klighd-core/src/views-rendering.tsx

```diff
@@ -964,40 +1111,41 @@ export function renderKRendering(kRendering: KRendering,
                     verticalAlignment: VerticalAlignment.CENTER
                 } as KVerticalAlignment
                 boundsAndTransformation.bounds = {
-                    x: calculateX(0, originalWidth, styles.kHorizontalAlignment, textWidth),
-                    y: originalHeight * 0.5,
-                    width: originalWidth,
-                    height: originalHeight
+                    x: calculateX(0, originalBounds.width, styles.kHorizontalAlignment, textWidth),
+                    y: originalBounds.height * 0.5,
+                    width: originalBounds.width,
+                    height: originalBounds.height
                 }
             } else {
                 // Offsets are already applied in the transformation, so set them to 0 here.
                 boundsAndTransformation.bounds = {
                     x: 0,
                     y: 0,
-                    width: originalWidth,
-                    height: originalHeight
+                    width: originalBounds.width,
+                    height: originalBounds.height
                 }
             }
-            if (providingRegion) {
-                // Store exact height of title text
-                providingRegion.regionTitleHeight = newHeight
-                providingRegion.regionTitleIndentation = newX
-            }
+
+            // Don't draw if the rendering is an empty KText
+            const isEmptyText = kRendering.type === K_TEXT && (kRendering as KText).text === ""
+
             // Draw white background for overlaying titles
-            if (context.depthMap && kRendering.isNodeTitle && ((providingRegion && providingRegion.detail === DetailLevel.FullDetails) || !providingRegion)
-                && kRendering.calculatedBounds && kRendering.calculatedBounds.height * context.viewport.zoom <= titleScalingFactorOption * kRendering.calculatedBounds.height
-                // Don't draw if the rendering is an empty KText
-                && (kRendering.type !== K_TEXT || (kRendering as KText).text !== "")) {
-                overlayRectangle = <rect x={0} y={0} width={originalWidth} height={originalHeight} fill="white" opacity="0.8" stroke="black"/>
+            if ((!providingRegion || providingRegion.detail === DetailLevel.FullDetails) && tooSmall && !isEmptyText) {
+                overlayRectangle = <rect x={0} y={0} width={originalBounds.width} height={originalBounds.height} fill="white" opacity="0.8" stroke="grey"/>
             }
+        } else {
+            context.pushEffectiveZoom(context.effectiveZoom)
         }
+    } else {
+        context.pushEffectiveZoom(context.effectiveZoom)
     }
     // Add the transformations to be able to positon the title correctly and above other elements
     context.positions[context.positions.length - 1] += (boundsAndTransformation?.transformation ?? "")
 
     let svgRendering: VNode
     switch (kRendering.type) {
         case K_CONTAINER_RENDERING: {
+            context.popEffectiveZoom()
```

Comment:
can't this be done outside of the switch to avoid duplication?

Reply:
the way i could think of would involve putting the switch in a try block with the pop in a corresponding finally block,
the problem is that some branches require the pop to not have happened yet, meaning we can't do it before the switch and some blocks return meaning we can't do it afterwards. Using try-finally could solve this as finally is executed even after return, but I am not sure whether such unusual control flow is warranted

Reply:
Instead of unusual control flow this could also have a "result" variable then taking the otherwise returned value, popping after the switch and return directly after that if the variable was written to. Better control flow imo and avoids this duplication.